lib/containers/container_images.pm

```diff
@@ -54,16 +54,44 @@ sub build_and_run_image {
     my $dir = "~/containerapp";
 
     # Setup the environment
-    prepare_img("$dir", $dockerfile, $base);
+    record_info('Downloading', "Dockerfile: containers/$dockerfile\nHTML: containers/index.html");
+    assert_script_run "mkdir -p $dir/BuildTest";
+    assert_script_run "curl -f -v " . data_url("containers/$dockerfile") . " > $dir/BuildTest/Dockerfile";
+    file_content_replace("$dir/BuildTest/Dockerfile", baseimage_var => $base) if defined $base;
+    assert_script_run "curl -f -v " . data_url('containers/index.html') . " > $dir/BuildTest/index.html";
 
     # Build the image
-    $buildah ? build_img("$dir", 'buildah') : build_img("$dir", $runtime);
+    assert_script_run("cd $dir");
+    if (!$buildah) {
+        # At least on publiccloud, this image pull can take long and occasinally fails due to network issues
+        assert_script_run("$runtime build -t myapp BuildTest", timeout => 300);
+        assert_script_run("$runtime images | grep myapp");
+    } ele {
```

Comment:
New invention? :P

https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12827

Reply:
Oh I didn't push this!?
Thank you for the fix.